source/common/http/conn_manager_impl.cc

```diff
@@ -1107,7 +1135,7 @@ void ConnectionManagerImpl::ActiveStream::encodeHeaders(ActiveStreamEncoderFilte
     (*continue_data_entry)->stopped_ = true;
     (*continue_data_entry)->continueEncoding();
   } else {
-    maybeEndEncode(end_stream);
+    maybeEndEncode(encoding_headers_only_ || end_stream);
```

Comment:
are we handling maybeEndEncode and maybeEndDecode the same for the header-only case?  If not, maybe worth a comment why we treat them differently?

Reply:
I think we're handling them differently just because they're called at different locations: maybeEndDecode is called *before* filters are evaluated, while maybeEndEncode is called after. I'll go over this and add some comments to make this easier to understand

Actually this makes me think we might not be calling endDecode properly, will investigate